desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py

```diff
@@ -219,6 +219,23 @@ def test_check_status(self):
       assert_equal(response['status'], 'available')
 
 
+  def test_explain(self):
+
+      with patch('notebook.connectors.sql_alchemy.SqlAlchemyApi._get_engine') as engine:
+        with patch('notebook.connectors.sql_alchemy.SqlAlchemyApi._get_session') as session:
+
+          snippet = {'statement': 'SELECT 1;'}
+          notebook = Mock()
+
+          output = ('id: 1\nselect_type: SIMPLE\ntable: server_cost\npartitions: None\ntype: ALL\npossible_keys: None\n' +
+            'key: None\nkey_len: None\nref: None\nrows: 6\nfiltered: 100.0\nExtra: None')
+          engine.execute().return_value = output
```

Comment:
Please read more some existing examples in the Hue code bases and MagicMock docs:

```
result = 'mocked data you want'
engine.execute = Mock(return_value=result)
```

https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/api_tests.py#L403
https://docs.python.org/3/library/unittest.mock.html